### ISSUE TITLE
Upgrade to jakarta Namespace

### ISSUE THREAD

**Issue by vlcheong** [mybatis/cdi#242]
Hi MyBatis,
Any plan to upgrade to jakarta namespace CDI library?

**Comment by ynleborg** [comment]
@hazendaz 
This site is down: http://xmlns.jcp.org/xml/ns/javaee/beans_1_1.xsd and my tests are failing due to 
"/target/test-classes/META-INF/beans.xml
Caused by: org.xml.sax.SAXParseException: The element type "META" must be terminated by the matching end-tag "</META>"."

is there a workaround for this (this in not the first time when oracle site is down)

**Comment by hazendaz** [comment]
site was up when I checked after seeing this so was intermittent.  However, this points out we should be using https instead of http for that for security reasons.  I'll update that shortly and given recent changes there may be enough to cut a new release.  This is unrelated to jakarta namespace though.

**Comment by george-trudeau** [comment]
+1 here, we've updated our environment to Jakarta EE 9.1 on Wildfly 26.
I've compiled a custom version of mybatis-cdi for our internal projects to use the jakarta.* namespace while we wait for an official release.

**Comment by hazendaz** [comment]
@george-trudeau As to what you compiled, did you just change to jakarta and skip tests given the underlying framework for testing doesn't yet support jakarta?  Also assume you tied to newer cdi/weld when doing so.

I'm trying to figure out how to approach this.  At the moment, need to get a release out and see any feedback as generics messed with a lot along with typing.  Tests and separate aquillian test noted in readme are confirmed working but need to confirm live code given its been 3 years since a release.  At that point, I'm thinking to just cut a branch so the master as it is at that point will become two lines to support.  For the master, I'll make weld-junit the primary test tool as noted in current code line and prevented due to legacy support.  Once that is done, I'll switch it over but not certain how to proceed with tests given the transaction manager used doesn't support jakarta yet.  Maybe we just change that to something else, haven't looked around but did note that at least the existing appears to recently have moved into github.  That could always come later I guess.

**Comment by george-trudeau** [comment]
Yes that's about all I did :
- upgraded jakarta.annotation-api to 2.0.0
- upgraded jakarta.transaction-api to 2.0.0
- added cdi-3.0 profile with jakarta.enterprise.cdi-api to 3.0.1
- dropped all other cdi profiles
- renamed javax.* to jakarta.* corresponding namespace

And unfortunately I had to skip the tests, since as you pointed out atomikos doesn't provide support for the new jakarta namespace.

I did look for an alternative implementation for the UserTransaction api, such as narayana, but I didn't spend much time trying to make it work.

**Comment by hazendaz** [comment]
@george-trudeau Thank you for the information!  I did get one last javax release out that I want to soak a bit as I reworked a lot in the generics area and type safety avoidance of <?>.  This info will help getting this effort started and great that my quick look was on par with your findings.  I'll try to get something out here around May 1st or so in support of jakarta.

**Comment by pangbuddy** [comment]
+1, with the release of Jakarta EE 10, a new JAVA era is coming, we need Mybatis-CDI to catch up with the team, thanks a lot!

**Comment by hazendaz** [comment]
I've added a jarkarta branch but tests don't currently work.  I haven't figured out the issue so if anyone wants to take a look at it or even build and give it a run, let me know how that goes.

**Comment by hazendaz** [comment]
Link to branch https://github.com/mybatis/cdi/tree/jakarta

**Comment by hazendaz** [comment]
More updates to jakarta branch were applied, it is basically complete but transaction manager used doesn't support jakarta for testing.  Either we need to migrate to something else or await support.

**Comment by hazendaz** [comment]
Code base is now migrated on tests to narayana.  Much thanks to https://github.com/TheOnlyAl!  I need to get mybatis core out this weekend for separate issue encountered with newer spring usage.  I plan then to cut once on javax so its up to date then will work on flip over to jakarta.  So possibly this all comes out this weekend.

**Comment by Wade-tech** [comment]
Hi  which mybatis version support jakarta.*?

**Comment by hazendaz** [comment]
Have not released yet, will try to get to it this weekend.  So many things to release for jakarta, sorry for slow progress.  Work here is done and if you can use snapshots believe it's on master already and in sonatype ready to test latest snapshot.  Do check our jakarta branch in off chance it wasn't merged to master yet.

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by hazendaz** [comment]
ok sorry my fault here, I had planned a final release on javax followed by merging in jakarta.  I had not done that as waiting on mybatis core which we had a defect on after my comments from feb 3.  I'm working both javax and jakarta release now.  Will be out shortly.  I'll close this ticket once done.

**Comment by hazendaz** [comment]
Release 1.1.4 is last for javaEE and supports mybatis 3.5.13 by default but many in general

Release 2.0.0 is jakarta.

Releases will be in central within few hours.

Closing ticket now as completed.  Thanks for patiently waiting this out.
